src/Uno.UI/UI/Xaml/UIElement.ThemeShadow.cs

```diff
@@ -0,0 +1,127 @@
+﻿using Uno.Disposables;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+#if __ANDROID__
+using Android.Graphics;
+#endif
+
+namespace Windows.UI.Xaml;
+
+public partial class UIElement
+{
+	private void UpdateShadow()
+	{
+		if (Shadow == null || Translation.Z <= 0)
+		{
+			UnsetShadow();
+		}
+		else
+		{
+			SetShadow();
+		}
+	}
+
+	partial void SetShadow();
+
+	partial void UnsetShadow();
+
+#if __ANDROID__
```

Comment:
Feels like this file should be split per platform :)

Reply:
Yea, I wasn't sure how to go about it - more files or single file with #if defs 😀